### PR TITLE
OSX Testing Workflow

### PR DESCRIPTION
**Pull request by arahlin**
Fixes for the testing github action with OSX.

Also disable running pipeline modules in a subprocess in python version > 3.7 due to limitations with pickling the module objects, so that tests with the latest python version pass.  Save the correct fix for this for a future PR.

### CODE REVIEW

File: core/python/modconstruct.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-from spt3g.core import G3Module, G3Pipeline, G3PipelineInfo, G3Frame, G3FrameType, G3Time, G3ModuleConfig, log_fatal
+from spt3g.core import G3Module, G3Pipeline, G3PipelineInfo, G3Frame, G3FrameType, G3Time, G3ModuleConfig, log_fatal, log_error
 try:
     from spt3g.core import multiprocess
     multiproc_avail = True
```
Comment (nwhitehorn) [inline core/python/modconstruct.py:4]: Why not make multiproc_avail be false for Python > 3.7?

File: core/tests/multiproc.py [hunk 2]
```diff
@@ -6,40 +6,38 @@
...
-	fr['time'] = core.G3Time(int(time.time()*core.G3Units.s))
-	fr['count'] = n
-	n += 1
+    global n
+    if fr.type != core.G3FrameType.Timepoint:
+        return
+    fr['time'] = core.G3Time(int(time.time()*core.G3Units.s))
+    fr['count'] = n
+    n += 1
 m = 0
```
Comment (nwhitehorn) [inline core/tests/multiproc.py:15]: Can you please undo the whitespace changes in this file?

Comment (arahlin) [reply core/tests/multiproc.py]: as you wish...

File: core/python/modconstruct.py (hunk 1 above)
Comment (arahlin) [reply core/python/modconstruct.py]: changed this to raise an `ImportError` in `core/python/multiprocessing.py` instead

**Pull request merged**